src/paradigma/heart_rate/tfd.py

```diff
@@ -170,15 +173,15 @@ def gen_analytic(x):
     x_fft = np.fft.fft(x)
 
     # Generate the analytic signal in the frequency domain
-    H = [1] + list(np.repeat(2, N-1)) + [1] + list(np.repeat(0, N-1))
+    H = np.concatenate(([1], np.repeat(2, N-1), [1], np.repeat(0, N-1)))
```

Comment:
You're not going to like this, but I've browsed online and the usage of `np.repeat` is not so efficient, so perhaps better to go back to the old format:

H = np.empty(2 * N)  # Preallocate an array of size 2*N
H[0] = 1            # First element
H[1:N] = 2          # Next N-1 elements
H[N] = 1            # Middle element
H[N+1:] = 0         # Last N-1 elements

But I'll leave that to you for now.

Reply:
Haha, I changed it back, since I experimented a bit and the allocated suggestion is twice as fast